javaagent-tooling/javaagent-tooling-java9/src/main/java/io/opentelemetry/javaagent/tooling/UnsafeInitializer.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.tooling;
+
+import io.opentelemetry.javaagent.bootstrap.AgentClassLoader;
+import java.lang.instrument.Instrumentation;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class UnsafeInitializer {
+  private static final Logger logger = LoggerFactory.getLogger(UnsafeInitializer.class);
+
+  static void initialize(Instrumentation instrumentation, ClassLoader classLoader) {
+    initialize(instrumentation, classLoader, true);
+  }
+
+  private static void initialize(
+      Instrumentation instrumentation, ClassLoader classLoader, boolean testUnsafePresent) {
+    Class<?> unsafeClass;
+    try {
+      unsafeClass = Class.forName("jdk.internal.misc.Unsafe");
+    } catch (ClassNotFoundException exception) {
+      return;
+    }
+
+    Map<String, Set<Module>> exports = new HashMap<>();
+    // expose jdk.internal.misc.Unsafe to our agent
+    // this is used to generate our replacement sun.misc.Unsafe and also by grpc/netty to call
+    // jdk.internal.misc.Unsafe.allocateUninitializedArray
+    exports.put(
+        unsafeClass.getPackage().getName(),
+        Collections.singleton(UnsafeInitializer.class.getModule()));
+    instrumentation.redefineModule(
```

Comment:
Should we do that before the `if (testUnsafePresent...` check? If there's a `sun.misc.Unsafe` present in the classpath do we need to expose the `jdk.internal` one to the agent explicitly?

Reply:
I noticed that grpc/netty that we use also desires access to some things that would require extra jvm arguments. Thought that might as well use the opportunity and let netty also use jvm internals. For example granting access to internal unsafe allows netty to allocate uninitialized arrays
```
17:58:55.427 INFO  i.o.s.LinuxTestContainerManager - STDERR: [otel.javaagent 2021-09-15 14:58:55:441 +0000] [main] DEBUG io.grpc.netty.shaded.io.netty.util.internal.PlatformDependent0 - jdk.internal.misc.Unsafe.allocateUninitializedArray(int): available
```
To be honest I have no idea whether this really is beneficial.

Reply:
Hmm, if exporter netty's able to use some of `jdk.internal` methods to make itself more efficient (I've no idea though, I just assume that 😅 ) that's a pretty good reason to expose `Unsafe` to the agent even if `sun.misc` is available.